plugins/modules/ec2_transit_gateway.py

```diff
@@ -376,36 +349,36 @@ def create_tgw(self, description):
         options["DnsSupport"] = self.enable_option_flag(self._module.params.get("dns_support"))
         options["MulticastSupport"] = self.enable_option_flag(self._module.params.get("multicast_support"))
 
+        params = {"Description": description, "Options": options}
+
         try:
-            response = self._connection.create_transit_gateway(Description=description, Options=options)
-        except (ClientError, BotoCoreError) as e:
-            self._module.fail_json_aws(e)
+            response = create_ec2_transit_gateway(self._connection, **params)
+        except AnsibleEC2Error as e:
+            self._module.fail_json_aws_error(e)
 
-        tgw_id = response["TransitGateway"]["TransitGatewayId"]
+        tgw_id = response["TransitGatewayId"]
 
         if wait:
             result = self.wait_for_status(wait_timeout=wait_timeout, tgw_id=tgw_id, status="available")
         else:
             result = self.get_matching_tgw(tgw_id=tgw_id)
 
-        self._results["msg"] = f" Transit gateway {result['transit_gateway_id']} created"
+        self._results["msg"] = f"Transit gateway {result['transit_gateway_id']} created"
 
         return result
 
-    def delete_tgw(self, tgw_id):
+    def delete_tgw(self, tgw_id: str) -> Dict[str, Any]:
         """
-        De;lete the transit gateway and optionally wait for status to become deleted
-
-        :param tgw_id: The id of the transit gateway
-        :return dict: transit gateway object
+        Delete the transit gateway and optionally wait for status to become deleted.
+        :param tgw_id: The id of the transit gateway.
+        :return: Transit gateway object.
         """
         wait = self._module.params.get("wait")
         wait_timeout = self._module.params.get("wait_timeout")
-
         try:
-            response = self._connection.delete_transit_gateway(TransitGatewayId=tgw_id)
-        except (ClientError, BotoCoreError) as e:
-            self._module.fail_json_aws(e)
+            response = delete_ec2_transit_gateway(self._connection, tgw_id)
```

Comment:
`response` is not used later in the function. Also [this function](https://github.com/ansible-collections/amazon.aws/pull/2325/files#diff-11d6a927433f45aa84de2f54fa0adf33e8094ebb417c75ea476ffe61b4e4f587R1768) will be handling the exception. Do we have to handle error in the modules file?